crates/teloxide/src/update_listeners/polling.rs

```diff
@@ -436,10 +438,22 @@ impl<B: Requester> Stream for PollingStream<'_, B> {
                     }
                 }
                 Err(err) => {
-                    // Prevents the CPU spike occuring at network connection lose: <https://github.com/teloxide/teloxide/issues/780>
-                    let backoff_strategy = &this.polling.backoff_strategy;
-                    this.eepy.set(Some(sleep(backoff_strategy(*this.error_count))));
-                    log::trace!("set {:?} reconnection delay", backoff_strategy(*this.error_count));
+                    /*
+                       In case of RetryAfter(..) error we pause the polling for the specified amount
+                       of seconds.
+
+                       Otherwise, in case of network connection lose (<https://github.com/teloxide/teloxide/issues/780>) we
+                       use the backoff strategy to prevent the high CPU usage due to multiple instant reconnections
+                    */
+                    if let Some(seconds) = err.retry_after() {
```

Comment:
I feel like this would be nicer as

```rust
let duration = match err.retry_after() { ... }
```

so that you don't have to duplicate other code (having the same log message is fine IMO)

Reply:
also I think you need to move change of `error_count` here & reset it in case of retry after